src/sema/eval.rs

```diff
@@ -196,3 +202,331 @@ pub fn eval_const_rational(
         )),
     }
 }
+
+//// Helper function that recurses the expression, and calls expression = eval_constants_in_expression().
+/// If expression is an arithmetic operation of two number literals, it will be stored in an expression list to later be evaluated by verify_expression_for_overflow().
+/// If the return is a diagnostic, store it in a diagnostics list.
+fn check_term_for_constant_overflow(expr: &Expression, params: &mut RecurseParams) -> bool {
+    match expr {
+        Expression::Add(..)
+        | Expression::Subtract(..)
+        | Expression::Multiply(..)
+        | Expression::Divide(..)
+        | Expression::Modulo(..)
+        | Expression::Power(..)
+        | Expression::ShiftLeft(..)
+        | Expression::ShiftRight(..)
+        | Expression::BitwiseAnd(..)
+        | Expression::BitwiseOr(..)
+        | Expression::BitwiseXor(..) => {
+            let expression = eval_constants_in_expression(expr.clone());
+            match expression {
```

Comment:
you can combine two lines to `match eval_constants_in_expression(expr.clone()) {`

You are still cloning here.